### PR TITLE
Use static variable for checking TF_CUDNN_USE_FRONTEND

### PR DESCRIPTION
**Pull request by kaixih**
This PR avoid checking TF_CUDNN_USE_FRONTEND every time we call the CudnnUseFrontend().

cc. @nluehr

### CODE REVIEW

**Review by timshen91 — APPROVED** [review]

File: tensorflow/core/util/use_cudnn.cc [hunk 1]
```diff
@@ -38,12 +38,15 @@ namespace tensorflow {
...
-  return value;
+  static bool result = [] {
+    bool value = false;
+    Status status = ReadBoolFromEnvVar("TF_CUDNN_USE_FRONTEND", false, &value);
+    if (!status.ok()) {
+      LOG(ERROR) << status;
+    }
+    return value;
+  }();
+  return result;
```
Comment (gzmkl) [inline tensorflow/core/util/use_cudnn.cc:49]: Better to move the #if logics inside the lambda function

Comment (kaixih) [reply tensorflow/core/util/use_cudnn.cc:50]: Yes, done.

**Review by timshen91 — APPROVED** [review]

**Comment by gbaned** [comment]
@kaixih  Can you please address Ubuntu Sanity errors? Thanks!

**Comment by kaixih** [comment]
I don't think the Ubuntu Sanity failed because of my change. From the log, I didn't find related info. @timshen91 can you advise?

**Pull request merged**